### PR TITLE
Cecil/payment-element-dotnet-update

### PR DESCRIPTION
**Pull request by cecilphillip**
### Summary
* Upgrade to .NET 6 (LTS)
* Enabled implicit usings
* Converted controller to Minimal APIs
* Removed Newtonsoft.Json dependency

### CODE REVIEW

**Review by Myskipper830 — APPROVED** [review]

File: payment-element/.env.exmaple [hunk 1]
```diff
@@ -0,0 +1,5 @@
+STRIPE_PUBLISHABLE_KEY=<replace-with-your-publishable-key>
```
Comment (cjavilla-stripe) [inline payment-element/.env.exmaple:1]: typo in the name of this file, should be `.env.example`

File: payment-element/server/dotnet/Program.cs [hunk 2]
```diff
@@ -1,30 +1,97 @@
...
+        );
+        app.Logger.LogInformation($"Webhook notification with type: {stripeEvent.Type} found for {stripeEvent.Id}");
+    }
+    catch (Exception e)
+    {
+        Console.WriteLine($"Something failed {e}");
+        return Results.BadRequest();
+    }
+
+    if (stripeEvent.Type == "checkout.session.completed")
```
Comment (cjavilla-stripe) [inline payment-element/server/dotnet/Program.cs:86]: For the PaymentElement integration, we want to listen for `payment_intent.succeeded` 

We may have missed this in other server implementations here. 

```cs
var paymentIntent = stripeEvent.Data.Object as Stripe.PaymentIntent;
```

I think we might also have constants for the event types in stripe-dotnet: https://github.com/stripe/stripe-dotnet/blob/master/src/Stripe.net/Constants/Events.cs#L558

Might be able to say something like:

```cs
if (stripeEvent.Type == Stripe.Events.PaymentIntentSucceeded)
```

File: payment-element/server/dotnet/server.csproj [hunk 3]
```diff
@@ -1,20 +1,11 @@
 <Project Sdk="Microsoft.NET.Sdk.Web">
-
   <PropertyGroup>
-    <TargetFrameworks>netcoreapp3.1;net5.0</TargetFrameworks>
+    <TargetFrameworks>net6.0</TargetFrameworks>
```
Comment (cjavilla-stripe) [inline payment-element/server/dotnet/server.csproj:3]: Wonder if we should update the README to show:

```bash
dotnet run Program.cs --framework=net6.0
```

Maybe we don't need the explicit version after this?

File: payment-element/.env.exmaple (hunk 1 above)
Comment (cecilphillip-stripe) [reply payment-element/.env.exmaple:1]: Good catch

File: payment-element/server/dotnet/Program.cs (hunk 2 above)
Comment (cecilphillip) [reply payment-element/server/dotnet/Program.cs:86]: Updated

File: payment-element/server/dotnet/server.csproj (hunk 3 above)
Comment (cecilphillip) [reply payment-element/server/dotnet/server.csproj:3]: we can should probably just target one framework. 6.0 is an LTS release

**Pull request merged**